### ISSUE TITLE
feature request: add a geo URL in the modal box for a displayed item, to allow users to navigate with their favorite app

### ISSUE THREAD

**Issue by govvin** [plepe/OpenStreetBrowser#143]
I just re-discovered OSB, and that currently, it's the best way to share links to people who may not use OSM. 

I used to share the OSM shortcut links, but people are confused because details of the features are not displayed (unless you give a link to the element) or they discover how to "query" the map. Not ideal for most people, and OSM provides a better option.

Using OSB, I could share a permalink like [this](https://www.openstreetbrowser.org/#map=17/14.65948/121.13419&categories=craft), and with a geo URL, it can allow the recipient to navigate to the location, and choose anyapp they have in their device.

A geo URL is platform agnostic, and should work in most platforms, and most apps.

**Comment by plepe** [comment]
The permalink feature is still there (on the bottom left), also you can just take the URL from the address bar.

I could convert the coordinates in the "Geometry"-toolbox to geo URLs? Or do you have a better idea?
![Screenshot 2021-08-05 at 16-59-22 OpenStreetBrowser](https://user-images.githubusercontent.com/249012/128372508-359f9693-5dff-4523-bc2f-4c7eebc3c1ff.png)

**Comment by govvin** [comment]
Well, what I had in mind is that it could be made friendlier, and more conveniently accessible, especially for mobile users with limited screen space?

So, right now, when a user clicks on a marker, the modal window shows this:

![image](https://user-images.githubusercontent.com/5821688/128440159-6a3e2b35-984a-4e60-ab87-1c586fe69480.png)

Perhaps, the geo URL can be activated by clicking on the address? That way,  they can still navigate to the given location no matter what device they're on, or whatever is their navigation app.

On mobile, the permalink option needs more scrolling right now:
![image](https://user-images.githubusercontent.com/5821688/128440491-7c3ff3fb-aa67-49a5-aca1-d6c2b5cde5e4.png)

**Comment by plepe** [comment]
Ah, now I understand what you have in mind.

There are some further considerations though:
* What's with POI which do not have an address?
* I could also add a 'Permalink' in the buttom link bar of the modal window?
* Should the 'Permalink' from the (bottom left) menu be visible all the time, resp. the whole menu? Should I move the menu to a tab (e.g. with the ☰ symbol)?

The view of the modal window is going to change "soon" (it is not totally finished yet): 
![Screenshot 2021-08-06 at 09-36-29 OpenStreetBrowser](https://user-images.githubusercontent.com/249012/128474403-7eea9e52-551a-41f2-9505-494823d7e9cd.png)

Read more about this in #132, and try it in my development version: https://xover.mud.at/~skunk/openstreetbrowser

**Comment by govvin** [comment]
I believe that even with POIs without any address information, they will have geographic coordinates, and wouldn't that be the minimum information needed for geo URIs?

**Comment by plepe** [comment]
What I mean is, that if a POI does not have an address, then the address is not shown, and therefore I can't make it a geo-URL. Of course, (almost) every object has coordinates (or it couldn't be displayed on the map).

Almost? Yes: in fact, you could have empty ways or relations in OpenStreetMap, but they do not really make sense.

**Comment by govvin** [comment]
Oh, I misunderstood you. Sorry, about that.

I'm not sure what the ideal placement is. I though address seem logical, but, as you pointed out, maybe not. 

As a hyperlink to the name, maybe?

**Comment by plepe** [comment]
I made all coordinate displays as geo-links (b10e448e5685b280d63c18273473f56faa78281a). You can find it under 'show details'. Do you think, this is sufficient?

**Comment by govvin** [comment]
It works as expected. Thank you. However,  I'm wondering if many people will be bothered to look for it?

I'm thinking of a casual, non-technical user who was given an OpenStreetBrowser link to a map or POI. We will not make an assumption about whether they have any app that can a geoURI.  In the pop-up, if a phone number is available, they can easily click it and make a call. If the address details are available, they see a text, but that may not be useful if they use another app (e.g. Google Maps in Android) for navigation.

They are then expected to go to "show details", and find the coordinates, and may not even know that it's a geoURI they may use. I think the geoURI link has more potential as clickable link for any address value. Do you think more people will that more convenient?

![image](https://user-images.githubusercontent.com/5821688/149609561-076719d7-b492-46ea-821c-4cb3e5f2c271.png)

**Comment by govvin** [comment]
P.S.

Apparently, on an iPhone, geoURI is not supported. Clicking on it doesn't appear to do anything.

**Comment by plepe** [comment]
I can add add it to the address too, but as I told you, many OSM items do not have an address. I could create an additional link on the bottom of the popup, between 'show details' and 'edit'. Like: 'geo-link'? I could add the coordinates (for nodes) or middle point (for lines) or the [center of mass](https://turfjs.org/docs/#centerOfMass) for areas.

I think I can't do anything, about iPhones not supporting 'geo:'-links, do I? I mean, if there's a different linking method, I could make it configurable (and even auto-detectable).

**Comment by govvin** [comment]
No, of course not.  I'm not an iOS user, but had the chance to try it earlier, and surprised that there's no support for it on that platform. The comment about iOS was just a general note, and not directed at you. I'm sorry if you thought otherwise.

Your suggested alternative sounds good tho. Thank you.
